src/hooks/useModalClose.ts

```diff
@@ -0,0 +1,20 @@
+'use client';
+
+import { useEffect } from 'react';
+
+export const useModalClose = (
+  ref: React.RefObject<HTMLElement>,
+  callback: () => void,
+) => {
+  const handleClickOutside = (e: MouseEvent) => {
+    if (ref.current && !ref.current.contains(e.target as Node)) {
+      callback();
+    }
+  };
+  useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  });
```

Comment:
의존성 배열에 아무것도 넣어주지 않은 이유가 있으실까요?? ㅎㅎㅎ 지금처럼 useEffect의 의존성 배열에 아무것도 전달하지 않는다면 컴포넌트가 렌더링 할 때마다 addEventListener가 실행되는 걸로 알고 있습니다 !!
1. 의존성 배열에 아무것도 전달하지 않을 시: 모든 렌더링 시에 실행
2. 빈 배열 `[]` 전달 시: 맨 처음 컴포넌트가 마운트 되었을 때 한 번만 실행된다.
3. 변수 전달 시, `[count]` : 맨 처음 마운트 되었을 때와, `count` state의 값이 바뀔 때만 실행된다.

그래서 현재 2번 로직이 현재 상황에 더 적합한 것 같아서, 의존성으로 빈 배열을 전달하는 게 어떨까 생각합니닷!😎

```suggestion
  useEffect(() => {
    document.addEventListener('mousedown', handleClickOutside);
    return () => {
      document.removeEventListener('mousedown', handleClickOutside);
    };
  }, []);
```

Reply:
위와 같은 방식으로 하면 오류가 발생해서 []를 제거했던건데, useCallback 이용해 의존성 배열에 [handleClickOutside] 추가하는 방식으로 변경했습니다!